### ISSUE TITLE
Bug: [@typescript-eslint/strict-boolean-expressions] falsey and truthy check is not working as expected

### ISSUE THREAD

**Issue by hasham7861** [typescript-eslint/typescript-eslint#8286]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=5.3.3&fileType=.ts&code=MYewdgzgLgBAhjAvDA5CgUOglgMxgCjgEoYBvdGGUSEAGwFMA6WkAc3xQhSPQF8g&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6ZfKsugIwHs%2BSAIZN6AD2LQUySnyaoMkRNGh9okADThseIqWQVqdBszZM%2BtWEyaIy0odEK1%2BgxCNqN8yofFpk%2BALbEDojoUMqq6lrYUOZ%2BchwidP5MACaU%2BLJMYUoqapDaAL4ghUA&tsconfig=&tokens=false

### Repro Code

```TypeScript
const x = 0

if (a) {
  console.log('s')
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/strict-boolean-expressions": "error",
  }
}
```

### tsconfig

```jsonc
undefined
```

### Expected Result

To throw a erro in if statement guard because it is doing falsey check

### Actual Result

no error thrown

**Comment by hasham7861** [comment]
Actually I found another rule in eslint called "@typescript-eslint/no-unnecessary-condition": "error" that takes care of it for me

<img width="770" alt="image" src="https://github.com/typescript-eslint/typescript-eslint/assets/16128567/505e2fc8-1619-4685-858f-34fa119de646">
